events/test/processor/index-test.js

```diff
@@ -0,0 +1,197 @@
+import { assert } from 'chai';
+import sinon from 'sinon';
+import Processor from '../../processor';
+import AWS from 'aws-sdk';
+import RecordException from '../../processor/types/exceptions';
+import Echo from '../../processor/types/echo';
+
+
+describe('Processor', () => {
+  describe('#constructor', () => {
+    it('creates a Kinesis client when one is not provided', () => {
+      const processor = new Processor({'Records': []});
+      assert.instanceOf(processor.kinesis, AWS.Kinesis);
+    });
+
+    it('sets event', () => {
+      const event = {'Records': []}
+      const processor = new Processor(event);
+      assert.equal(processor.event, event);
```

Comment:
if event is an object you are going to want http://chaijs.com/api/assert/#method_deepequal